hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -92,7 +93,8 @@
  * </ul>
  */
 @InterfaceAudience.Private
-public class ReplicationSourceManager implements ReplicationListener {
+public class ReplicationSourceManager implements ReplicationListener,
+  ReplicationSourceOverallController {
```

Comment:
Yeah, if this implements Controller, you have to pass it in to ReplicationSource #Init as a distinct arg?